src/common/sdpa_test_iface.cpp

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+* Copyright 2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "c_types_map.hpp"
+#include "common/primitive_desc_iface.hpp"
+#include "common/sdpa_pd.hpp"
+#include "common/sdpa_types.hpp"
+#include "common/sdpa_utils.hpp"
+#include "opdesc.hpp"
+
+using dnnl::impl::status_t;
+using namespace dnnl::impl;
+
+dnnl_status_t DNNL_API dnnl_sdpa_primitive_desc_create(
+        dnnl_primitive_desc_t *primitive_desc_iface, dnnl_engine_t engine,
+        const_dnnl_memory_desc_t query_desc, const_dnnl_memory_desc_t key_desc,
+        const_dnnl_memory_desc_t value_desc, const_dnnl_memory_desc_t dst_desc,
+        const_dnnl_memory_desc_t mask_desc, dnnl_data_type_t scale_dt,
+        bool invert_scale, dnnl_dim_t kv_head_number, bool causal_mask,
+        const_dnnl_primitive_attr_t attr, const_dnnl_primitive_attr_t kq_attr,
+        const_dnnl_primitive_attr_t vs_attr) {
```

Comment:
Minor: usually, the types for interface implementation are internal (the types that don't have "dnnl_" prefix).